### PR TITLE
[RNMobile] Add paused media state handlers to Android bridge code 

### PR DESCRIPTION
**Pull request by derekblank**
## What?
Adds Android bridge handlers for `MEDIA_UPLOAD_STATE_PAUSED`.

Branched from `rnmobile/media-upload-progress-connection-subscription`.

References:
* https://github.com/WordPress/gutenberg/pull/57476
* https://github.com/wordpress-mobile/gutenberg-mobile/pull/6499
* https://github.com/wordpress-mobile/WordPress-Android/pull/19867

## Why?
To allow the media upload progress bar status to persist when an upload is paused. 

More info:

* https://github.com/WordPress/gutenberg/pull/56937

## How?
- Adds `onMediaFileUploadPaused` handler
- Adds `progress` values to `onMediaFileUploadFailed` and `onUploadMediaFileClear` handlers for consistency and to reduce the effort to access these values in the future, if needed.

## Testing Instructions
N/A currently, WIP

### CODE REVIEW

File: packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergBridgeJS2Parent.java [hunk 1]
```diff
@@ -28,17 +28,19 @@ interface MediaSelectedCallback {
...
+        void onMediaFileUploadPaused(int mediaId, float progress);
     }
 
     interface MediaSaveEventEmitter {
-        void onSaveMediaFileClear(String mediaId);
+        void onSaveMediaFileClear(String mediaId, float progress);
         void onMediaFileSaveProgress(String mediaId, float progress);
         void onMediaFileSaveSucceeded(String mediaId, String mediaUrl);
-        void onMediaFileSaveFailed(String mediaId);
+        void onMediaFileSaveFailed(String mediaId, float progress);
```
Comment (dcalhoun) [inline packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergBridgeJS2Parent.java:42]: Were the `float progress` additions in the name of consistency? I noted the Android implementation has a dedicated progress event/method. 

https://github.com/WordPress/gutenberg/blob/0aabb54e931fb4652bbe22d1fc03764dc69b55c2/packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java#L1145-L1147

I.e., I wondered if Android is currently implemented in a way that negates the need to pass progress to the other methods. Should we avoid adding these `float progress`?

iOS is implemented very differently and does not have a dedicated progress event/method. iOS relies upon a enum state that is passed to a single [media observer method](https://github.com/wordpress-mobile/WordPress-iOS/blob/517584306972888d9ddc912f274d0d09046cde46/WordPress/Classes/ViewRelated/Gutenberg/GutenbergMediaInserterHelper.swift#L198-L266) with a switch statement that communicates different states/progresses to the JavaScript app. The single method can receive a [progress event](https://github.com/wordpress-mobile/WordPress-iOS/blob/517584306972888d9ddc912f274d0d09046cde46/WordPress/Classes/ViewRelated/Gutenberg/GutenbergMediaInserterHelper.swift#L262-L264) with a value.

File: packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/WPAndroidGlue/DeferredEventEmitter.java [hunk 2]
```diff
@@ -176,14 +176,19 @@ public void onMediaFileUploadSucceeded(int mediaId, String mediaUrl, int mediaSe
...
     @Override
-    public void onMediaFileUploadFailed(int mediaId) {
-        setMediaFileUploadDataInJS(MEDIA_UPLOAD_STATE_FAILED, mediaId, null, 0);
+    public void onMediaFileUploadFailed(int mediaId, float progress) {
+        setMediaFileUploadDataInJS(MEDIA_UPLOAD_STATE_FAILED, mediaId, null, progress);
+    }
+
+    @Override
+    public void onMediaFileUploadPaused(int mediaId, float progress) {
+        setMediaFileUploadDataInJS(MEDIA_UPLOAD_STATE_PAUSED, mediaId, null, progress);
```
Comment (dcalhoun) [inline packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/WPAndroidGlue/DeferredEventEmitter.java:187]: I believe `MEDIA_UPLOAD_STATE_PAUSED` needs to be defined as `11` atop this file. Currently it is undefined.

**Review by dcalhoun — COMMENTED** [review]
Thank you for starting the investigation into this. I left a few questions on which I'd love your thoughts. 

If you decide to move these changes (or others) forward in some manner, feel free to merge the PRs into `rnmobile/android-media-paused` as you see fit.

**Comment by derekblank** [comment]
Closing, rethinking this approach to focus more on the paused state (instead of passing progress to the failed state) and to better align with the [iOS implementation](https://github.com/wordpress-mobile/WordPress-iOS/pull/22282/files) by reducing boilerplate handlers.

**Pull request closed without merging**